Source/Fuse.Controls.Navigation/NavigationControl.Pages.uno

```diff
@@ -95,13 +95,13 @@ namespace Fuse.Controls
 				
 			var trans = NavigationGotoMode.Transition;
 			
-			Visual v;
-			(this as IRouterOutlet).Goto( ref path, ref param, trans, op, "", out v );
-			if (v != null)
+			RouterPage rPage = new RouterPage{ Path = path, Parameter = param };
+			(this as IRouterOutlet).Goto( rPage, trans, op, "" );
```

Comment:
I know that the code already did this, so just take this as a thought along the road:

This should use a real cast instead of `as`. If the cast would fail, the cast would return null, which would give us a NRE. An InvalidCastException would be much more helpful in such a case. A real cast conveys the information that this isn't supposed to fail much clearer than `as` does.

Reply:
But reading the code further, how *is* `NavigationControl` a `IRouterOutlet`? I can't see either it nor any of it's base-classes implementing `IRouterOutlet`... What am I missing?

Reply:
Yeah, I just never thought of this, copying what the code already did. It seems weird as a whole that I need to cast at all to call functions on `this`, but oh well.

Reply:
Both of it's derived classed implement `IRouterOutlet`.

It appears to be a C# (not necessarily Uno) problem that it's difficult to get this interface on the base class when you have derived classes implementing it.

Reply:
OK, so we depend on derived classes to implement IRouterOutlet?

Reply:
Yes.